db/sql/00_msar.sql

```diff
@@ -5164,6 +5220,59 @@ END;
 $$ LANGUAGE SQL STABLE RETURNS NULL ON NULL INPUT;
 
 
+CREATE OR REPLACE FUNCTION msar.build_expr_and_ctes_for_enriched_results(
```

Comment:
I don't think blobbing all this logic into this function is a good idea.
- It obscures the logic and what the inputs/outputs are from each spot the various returned pieces are used
- It returns a complex object that isn't useful unless you look at the function's source code: It's not possible from the signature to determine what the function actually returns
- It's not documented, exacerbating the above problem.

Reply:
I added a commit to improve the name of this function and some comments: https://github.com/mathesar-foundation/mathesar/pull/4090/commits/56481fdd6a6f6a52b8ea93e73f1ad166e38f1e21.

I hope this should alleviate some of the concerns. However, I'm open to refactoring this later if required.